lightning/src/offers/invoice_request.rs

```diff
@@ -81,18 +83,54 @@ const SIGNATURE_TAG: &'static str = concat!("lightning", "invoice_request", "sig
 /// [module-level documentation]: self
 pub struct InvoiceRequestBuilder<'a> {
 	offer: &'a Offer,
-	invoice_request: InvoiceRequestContents,
+	invoice_request: InvoiceRequestContentsWithoutPayerId,
+	metadata: Metadata,
+	payer_id: Option<PublicKey>,
 }
 
 impl<'a> InvoiceRequestBuilder<'a> {
 	pub(super) fn new(offer: &'a Offer, metadata: Vec<u8>, payer_id: PublicKey) -> Self {
 		Self {
 			offer,
-			invoice_request: InvoiceRequestContents {
+			invoice_request: InvoiceRequestContentsWithoutPayerId {
 				payer: PayerContents(metadata), offer: offer.contents.clone(), chain: None,
 				amount_msats: None, features: InvoiceRequestFeatures::empty(), quantity: None,
-				payer_id, payer_note: None,
+				payer_note: None,
+			},
+			metadata: Metadata::Empty,
```

Comment:
It's weird that we have two metadata fields in `InvoiceRequestBuilder` that aren't super explicitly disambiguated. Not sure if it'd be possible to pare down to one